src/LowLevelMotionController.cpp

```diff
@@ -136,6 +142,16 @@ int main(int argc, char **argv)
 
     while (ros::ok())
     {
+        // Check the safety client before updating anything
+        ROS_ASSERT_MSG(!safety_client.isFatalActive(), "low_level_motion: fatal event from safety");
+
+        if(safety_client.isSafetyActive())
+        {
+            // Override whatever the Acceleration Planner wants to do and attempt to land
+            geometry_msgs::Twist twist;
+            twist.linear.z = -0.2; // Try to descend
+            quadController.setTargetVelocity(twist);
```

Comment:
I agree. I dislike the idea of modifying the quadvelocity controller to be aware of it's input source that's far outside of the responsibility of the class.

After thinking about it a bit, I realized that the structure of the acceleration planner doesn't match the rest of the program. It shouldn't be responsible for executing it's update functions periodically like it is. (This also means the correct velocities are not always being sent to the velocity controller because of the difference in when the velocities are generated and the velocity controller updating). I will refactor it to be updated from the main program loop, this will double in making the intentions of the safety functionality clear and refrain from doing unnecessary, inaccurate calculations.